api.js

```diff
@@ -61,19 +62,44 @@ Api.prototype._reset = function () {
 	this.base = '';
 };
 
-Api.prototype._runFile = function (file) {
+Api.prototype._runFile = function (file, onForkStarting) {
```

Comment:
This is a weird combination of callbacks and promises. Can we clean it up to just use promises?